docs/PublishingGuide.md

```diff
@@ -0,0 +1,67 @@
+## Using Travis to automate publishing of your report to GitHub Pages
+
+[Travis-CI](https://travis-ci.org/) enables you to automate RepoSense report generation and publish the report online to [GitHub Pages](https://pages.github.com/) for free.
+
+1. Fork publish-RepoSense repository using this [link](https://github.com/RepoSense/publish-RepoSense/fork)
+1. Follow this [guide](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/) to generate a `Personal access token` on GitHub for report publishing <br/>
+*Remember to **copy it** and you would **only require** `public_repo` permission*
+1. Sign up and login to [Travis-CI](https://travis-ci.org/)
+1. Go to [your account](https://travis-ci.org/account/repositories), click on **Sync account** to fetch all your repositories into Travis-CI
+1. Go to [your publish-RepoSense fork in Travis-CI](https://travis-ci.org/search/publish-RepoSense/), under **Current** tab click on **Activate repository**
+1. In the same page, click on **More options** on the right then access **Settings**
+![Travis-CI Dashboard](images/publishingguide-travissetting.jpg "Travis-CI Dashboard")
+1. Under **Environment Variables**, name a variable as `GITHUB_TOKEN` and paste the `Personal access token` to its value field; then click **Add** <br/>
+*Ensure that the `Display value in build log` is* **switched off** for security reasons
+![Travis-CI Environment Variable](images/publishingguide-githubtoken.jpg "Travis-CI Environment Variable")
+1. Edit [run.sh](../../../../publish-RepoSense/edit/master/run.sh), [repo-config.csv](../../../../publish-RepoSense/edit/master/configs/repo-config.csv) and [author-config.csv](../../../../publish-RepoSense/edit/master/configs/author-config.csv) to customize the command line parameters or repositories to be analyzed <br/>
+*Read our [User Guide](UserGuide.md#customizing-the-analysis) for more information*
+1. To access your site, go to [the settings of your fork in GitHub, under **GitHub Pages** section](../../../../publish-RepoSense/settings), look for `Your site is published at [LINK]` <br/>
```

Comment:
Sorry, meant to say the link here (GitHub pages) is not working 😓

Reply:
Works for me. Can try accessing it from https://github.com/yong24s/RepoSense/blob/master/docs/PublishingGuide.md

Reply:
Still not working for me  (404 error) 🤔 
Directs me to https://github.com/reposense/publish-RepoSense/settings, https://github.com/yong24s/publish-RepoSense/settings which gives 404

Reply:
😱 

The intention of these links are to help users navigate to the file in their fork repository.

I just realize that they don't do that now as it still refers to the upstream repository instead of the user's fork.

@damithc @eugenepeh 
Should we move this guide to `reposense/publish-RepoSense`?

Reply:
> Should we move this guide to `reposense/publish-RepoSense`?

Doesn't sound right, as the starter repo should not contain stuff that is not needed for it to work. Just revise the doc to remove hyper link?